src/ai/backend/gateway/session.py

```diff
@@ -689,6 +689,16 @@ def interrupt_wait(ctx: Any, event_name: str, agent_id: AgentId,
     await app['registry'].handle_stats(agent_id, kern_stats)
 
 
+async def handle_kernel_log(app: web.Application, agent_id: AgentId, event_name: str,
+                            raw_kernel_id: str, log_str: str):
```

Comment:
As commented in the agent PR, let's use a separate Redis key to store/retrieve the log data as list of Redis strings.